src/components/PageResultsNew/TestSuitesAccordion.tsx

```diff
@@ -0,0 +1,194 @@
+/*
+ * This file is part of ciboard
+ *
+ * Copyright (c) 2023 Matěj Grabovský <mgrabovs@redhat.com>
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+import * as _ from 'lodash';
+import { useState } from 'react';
+import {
+    Accordion,
+    AccordionContent,
+    AccordionItem,
+    AccordionToggle,
+    Alert,
+    Flex,
+    FlexItem,
+    Label,
+    Title,
+} from '@patternfly/react-core';
+import { MicrochipIcon, OutlinedClockIcon } from '@patternfly/react-icons';
+import { TableComposable, Tbody, Td, Tr } from '@patternfly/react-table';
+import moment from 'moment';
+import update from 'immutability-helper';
+
+import { TestCase, TestSuite } from './types';
+import { mkSeparatedList } from '../../utils/artifactsTable';
+import { renderStatusIcon } from '../../utils/artifactUtils';
+
+function humanReadableTime(seconds: number) {
+    // TODO: Migrate away from the moment library.
+    const duration = moment.duration(seconds, 'seconds');
+    if (duration.hours() >= 1)
+        return duration.format('hh:mm:ss', { trim: false });
+    return duration.format('mm:ss', { trim: false });
+}
+
+interface ArchitectureLabelProps {
+    architecture?: string;
+}
+
+const ArchitectureLabel: React.FC<ArchitectureLabelProps> = ({
+    architecture,
+}) => {
+    if (_.isEmpty(architecture)) return null;
+    return (
+        <Label
+            icon={<MicrochipIcon className="pf-u-color-200" />}
+            isCompact
+            title={`This test was run on the ${architecture} machine architecture`}
+        >
+            {architecture}
+        </Label>
+    );
+};
+
+interface LogsLinksProps {
+    logs: string[];
+}
+
+function LogsLinks(props: LogsLinksProps) {
+    const { logs } = props;
+    return (
+        <div className="pf-u-font-size-sm">
+            Logs:{' '}
+            {mkSeparatedList(logs.map((name) => <a href="#">{name}</a>, ', '))}
```

Comment:
:-D Maybe move out of rendering this?
This creates a list of .... hard to read.

Reply:
Refactored.